app/bundles/LeadBundle/Entity/LeadFieldRepository.php

```diff
@@ -93,6 +94,26 @@ public function getFields(): array
         return array_column($results, null, 'alias');
     }
 
+    /**
+     * @return string[]
+     */
+    public function getSearchableFieldAliases(string $object = null): array
+    {
+        $fq = $this->createQueryBuilder($this->getTableAlias());
+        $fq->select($this->getTableAlias().'.alias')
+            ->andWhere($fq->expr()->eq($this->getTableAlias().'.isSearchable', true))
+            ->andWhere($fq->expr()->eq($this->getTableAlias().'.isPublished', true));
```

Comment:
Shall we also somehow enforce that if a field is `searchable`, it should also be `isIndex`?

Reply:
I had the same thought and did not think it through while implementing. I'll get the PR updated; thanks, @escopecz.